exchanges/okx/okx_test.go

```diff
@@ -3369,372 +2930,951 @@ func TestGetHistoricalFundingRates(t *testing.T) {
-	if err != nil {
-		t.Error(err)
-	}
-	if is {
-		t.Error("expected false")
-	}
+	assert.NoError(t, err)
+	assert.False(t, is, "expected false")
 
 	cp, err := currency.NewPairFromString("BTC-USD-SWAP")
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, err)
 	is, err = ok.IsPerpetualFutureCurrency(asset.PerpetualSwap, cp)
-	if err != nil {
-		t.Error(err)
-	}
-	if !is {
-		t.Error("expected true")
-	}
+	assert.NoError(t, err)
+	assert.True(t, is, "expected true")
 }
 
 func TestGetAssetsFromInstrumentTypeOrID(t *testing.T) {
 	t.Parallel()
 	_, err := ok.GetAssetsFromInstrumentTypeOrID("", "")
-	if !errors.Is(err, errEmptyArgument) {
-		t.Error(err)
-	}
+	assert.True(t, errors.Is(err, errEmptyArgument), err)
 
 	assets, err := ok.GetAssetsFromInstrumentTypeOrID("SPOT", "")
-	if !errors.Is(err, nil) {
-		t.Error(err)
-	}
-	if len(assets) != 1 {
-		t.Errorf("received %v expected %v", len(assets), 1)
-	}
-	if assets[0] != asset.Spot {
-		t.Errorf("received %v expected %v", assets[0], asset.Spot)
-	}
+	assert.NoError(t, err)
+	assert.Falsef(t, len(assets) != 1, "received %v expected %v", len(assets), 1)
+	assert.Falsef(t, assets[0] != asset.Spot, "received %v expected %v", assets[0], asset.Spot)
 
 	assets, err = ok.GetAssetsFromInstrumentTypeOrID("", ok.CurrencyPairs.Pairs[asset.Futures].Enabled[0].String())
-	if !errors.Is(err, nil) {
-		t.Error(err)
-	}
-	if len(assets) != 1 {
-		t.Errorf("received %v expected %v", len(assets), 1)
-	}
-	if assets[0] != asset.Futures {
-		t.Errorf("received %v expected %v", assets[0], asset.Futures)
-	}
+	assert.NoError(t, err)
+	assert.False(t, len(assets) != 1, "received %v expected %v", len(assets), 1)
+	assert.Falsef(t, assets[0] != asset.Futures, "received %v expected %v", assets[0], asset.Futures)
 
 	assets, err = ok.GetAssetsFromInstrumentTypeOrID("", ok.CurrencyPairs.Pairs[asset.PerpetualSwap].Enabled[0].String())
-	if !errors.Is(err, nil) {
-		t.Error(err)
-	}
-	if len(assets) != 1 {
-		t.Errorf("received %v expected %v", len(assets), 1)
-	}
-	if assets[0] != asset.PerpetualSwap {
-		t.Errorf("received %v expected %v", assets[0], asset.PerpetualSwap)
-	}
+	assert.NoError(t, err)
+	assert.Falsef(t, len(assets) != 1, "received %v expected %v", len(assets), 1)
+	assert.Falsef(t, assets[0] != asset.PerpetualSwap, "received %v expected %v", assets[0], asset.PerpetualSwap)
 
 	_, err = ok.GetAssetsFromInstrumentTypeOrID("", "test")
-	if !errors.Is(err, currency.ErrCurrencyNotSupported) {
-		t.Error(err)
-	}
+	assert.True(t, errors.Is(err, currency.ErrCurrencyNotSupported), err)
 
 	_, err = ok.GetAssetsFromInstrumentTypeOrID("", "test-test")
-	if !errors.Is(err, asset.ErrNotSupported) {
-		t.Error(err)
-	}
+	assert.True(t, errors.Is(err, asset.ErrNotSupported), err)
 
 	assets, err = ok.GetAssetsFromInstrumentTypeOrID("", ok.CurrencyPairs.Pairs[asset.Margin].Enabled[0].String())
-	if !errors.Is(err, nil) {
-		t.Error(err)
-	}
+	assert.True(t, errors.Is(err, nil), err)
```

Comment:
`assert.ErrorIs(t, err, nil)` and across this file and your defined error aacross this file.